geopandas/geoseries.py

```diff
@@ -619,6 +617,14 @@ def to_file(
     def _constructor(self):
         return _geoseries_constructor_with_fallback
 
+    def _constructor_from_mgr(self, mgr, axes):
+        assert isinstance(mgr, SingleBlockManager)
+
+        if not isinstance(mgr.blocks[0].dtype, GeometryDtype):
+            return Series._from_mgr(mgr, axes)
```

Comment:
@jbrockmendel What do you think is the best way to override this as a subclass? Does this look reasonable? Another option I was initially thinking about was to do something like `super()._constructor_from_mgr(mgr, axes)` (but that doesn't work because `self` is still the subclass, and you can't do `Series._constructor_from_mgr` because it's an instance method)

Reply:
> I guess that is slower than this?

Yeah, that then also goes through the `Series(..)` constructor

Reply:
how much slower?  i have a hard time imagining it as being a real pain point